packages/react-component-library/src/components/DatePickerE/useDatePickerEReducer.ts

```diff
@@ -0,0 +1,103 @@
+import { isEqual, isValid } from 'date-fns'
+import React, { useEffect, useReducer } from 'react'
+
+import { formatDatesForInput } from './formatDatesForInput'
+import { DatePickerEAction, DatePickerEState } from './types'
+
+function init({
+  from,
+  to,
+  datePickerFormat,
+}: Omit<DatePickerEState, 'inputValue'>) {
+  return {
+    from,
+    to,
+    datePickerFormat,
+    inputValue: formatDatesForInput(from, to, datePickerFormat),
+  }
+}
+
+function reducer(state: DatePickerEState, action: DatePickerEAction) {
+  switch (action.type) {
+    case 'reset':
+      return init(action.data)
+    case 'update':
+      return { ...state, ...action.data }
+    case 'refreshInputValue':
+      if (state.from && !isValid(state.from)) {
+        return state
+      }
+
+      return {
+        ...state,
+        inputValue: formatDatesForInput(
+          state.from,
+          state.to,
+          state.datePickerFormat
+        ),
+      }
+    default:
+      throw new Error('Unknown reducer action type')
```

Comment:
I wonder if we change the error message elsewhere to be consistent across the code base:

https://github.com/defencedigital/mod-uk-design-system/blob/12e4eda364b81fd657257be1417334f0c068038d/packages/react-component-library/src/components/Timeline/context/reducer.ts#L145